tests/drivers/display/display_check/src/main.c

```diff
@@ -0,0 +1,383 @@
+		buf_size *= 3;
+		break;
+	case PIXEL_FORMAT_RGB_565:
+		bg_color = 0xFFu;
+		fill_buffer_fnc = fill_buffer_rgb565;
+		buf_size *= 2;
+		break;
+	case PIXEL_FORMAT_BGR_565:
+		bg_color = 0xFFu;
+		fill_buffer_fnc = fill_buffer_bgr565;
+		buf_size *= 2;
+		break;
+	case PIXEL_FORMAT_L_8:
+		bg_color = 0xFFu;
+		fill_buffer_fnc = fill_buffer_l_8;
+		break;
+	case PIXEL_FORMAT_MONO01:
+		bg_color = 0xFFu;
+		fill_buffer_fnc = fill_buffer_mono01;
+		buf_size = DIV_ROUND_UP(DIV_ROUND_UP(
+			buf_size, NUM_BITS(uint8_t)), sizeof(uint8_t));
+		break;
+	case PIXEL_FORMAT_MONO10:
+		bg_color = 0x00u;
+		fill_buffer_fnc = fill_buffer_mono10;
+		buf_size = DIV_ROUND_UP(DIV_ROUND_UP(
+			buf_size, NUM_BITS(uint8_t)), sizeof(uint8_t));
+		break;
+	default:
+		LOG_ERR("Unsupported pixel format. Aborting test.");
+#ifdef CONFIG_ARCH_POSIX
+		posix_exit_main(1);
+#else
+		return 0;
+#endif
+	}
+
+	buf = k_malloc(buf_size);
+
+	if (buf == NULL) {
+		LOG_ERR("Could not allocate memory. Aborting test.");
+#ifdef CONFIG_ARCH_POSIX
+		posix_exit_main(1);
+#else
+		return 0;
+#endif
+	}
+
+	(void)memset(buf, bg_color, buf_size);
+
+	buf_desc.buf_size = buf_size;
+	buf_desc.pitch = capabilities.x_resolution;
+	buf_desc.width = capabilities.x_resolution;
+	buf_desc.height = h_step;
+
+	/*
+	 * The following writes will only render parts of the image,
+	 * so turn this option on.
+	 * This allows double-buffered displays to hold the pixels
+	 * back until the image is complete.
+	 */
+	buf_desc.frame_incomplete = true;
+
+	for (int idx = 0; idx < capabilities.y_resolution; idx += h_step) {
+		/*
+		 * Tweaking the height value not to draw outside of the display.
+		 * It is required when using a monochrome display whose vertical
+		 * resolution can not be divided by 8.
+		 */
+		if ((capabilities.y_resolution - idx) < h_step) {
+			buf_desc.height = (capabilities.y_resolution - idx);
+		}
+		display_write(display_dev, 0, idx, &buf_desc, buf);
+	}
+
+	buf_desc.pitch = rect_w;
+	buf_desc.width = rect_w;
+	buf_desc.height = rect_h;
+
+	fill_buffer_fnc(TOP_LEFT, 0, buf, buf_size);
+	x = 0;
+	y = 0;
+	display_write(display_dev, x, y, &buf_desc, buf);
+
+	fill_buffer_fnc(TOP_RIGHT, 0, buf, buf_size);
+	x = capabilities.x_resolution - rect_w;
+	y = 0;
+	display_write(display_dev, x, y, &buf_desc, buf);
+
+	/*
+	 * This is the last write of the frame, so turn this off.
+	 * Double-buffered displays will now present the new image
+	 * to the user.
+	 */
+	buf_desc.frame_incomplete = false;
+
+	fill_buffer_fnc(BOTTOM_RIGHT, 0, buf, buf_size);
+	x = capabilities.x_resolution - rect_w;
+	y = capabilities.y_resolution - rect_h;
+	display_write(display_dev, x, y, &buf_desc, buf);
+
+	display_blanking_off(display_dev);
+
+	grey_count = 0;
+	x = 0;
+	y = capabilities.y_resolution - rect_h;
+
+	LOG_INF("Display starts");
+	while (1) {
+		fill_buffer_fnc(BOTTOM_LEFT, grey_count, buf, buf_size);
+		display_write(display_dev, x, y, &buf_desc, buf);
+		++grey_count;
```

Comment:
If a corner of the display is constantly changing. Maybe this gives some area a random color?

This looks like the display sample was imported, maybe the `grey_count` argument can be removed to simplify.